*WebAPIsample프로젝트 삭제

raymondlee-nsuslab opened the pull request:
*WebAPIsample프로젝트 삭제

raymondlee-nsuslab committed:
*WebAPIsample프로젝트 삭제

raymondlee-nsuslab committed:
api 참조 클래스 라이브러리 추가

raymondlee-nsuslab committed:
*Server to Server api 호출입니다.

raymondlee-nsuslab reviewed (commented):
* Sever to Server api

greatyeon commented on ApiReferences/StudentReference.cs:
```diff
@@ -0,0 +1,119 @@
+                client.BaseAddress = new Uri("http://localhost/");
+                client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+                var resp = client.GetAsync("students/gettitles").Result;
```
약어는 가급적 쓰지 말아요...

greatyeon commented on ApiReferences/StudentReference.cs:
```diff
@@ -0,0 +1,119 @@
+                var resp = client.PostAsJsonAsync("students/getschools",students).Result;
+                if (!resp.IsSuccessStatusCode)
+                {
+                    var respErr = new HttpResponseMessage(HttpStatusCode.NotFound)
```
여기 위아래도 마찬가지로 약어 쓰지 말아요

greatyeon commented on ApiReferences/StudentReference.cs:
```diff
@@ -0,0 +1,119 @@
+        {
+            using (var client = new HttpClient())
+            {
+                client.BaseAddress = new Uri("http://localhost/");
```
혹시 사이먼이 한거 보고 했어요?
똑같은데...

greatyeon commented on ApiReferences/StudentReference.cs:
```diff
@@ -0,0 +1,119 @@
+
+namespace ApiReferences
+{
+    public class StudentReference
```
이거 이름이 좀... 다른 이름으로 고민해봐요...

greatyeon commented on ApiReferences/StudentsModels.cs:
```diff
@@ -0,0 +1,39 @@
+        public decimal? Grade { get; set; }
+    }
+
+    public class GetRequestStudents
```
GetStudentsRequet 가 좋겠어요

greatyeon commented on ApiReferences/StudentsModels.cs:
```diff
@@ -0,0 +1,39 @@
+
+namespace ApiReferences
+{
+    public class StudentsModels
```
Student 로 해도 되고...
이게 저번에 다 고친 것들 아닌가? 놓쳤나?

greatyeon commented on ApiReferences/StudentsModels.cs:
```diff
@@ -0,0 +1,39 @@
+{
+    public class StudentsModels
+    {
+        public int StudentModelsID { get; set; }
```
Models 이런건 들어가야 알 수 있는거예요?

greatyeon commented on ApiReferences/StudentsModels.cs:
```diff
@@ -0,0 +1,39 @@
+        public string Draw { get; set; }
+        public int RecordsFiltered { get; set; }
+        public int RecordTotal { get; set; }
+        public List<StudentsModels> StudentsModelses { get; set; }
```
Students 가 되면 될 것 같고...

greatyeon commented on ApiReferences/StudentsModels.cs:
```diff
@@ -0,0 +1,39 @@
+        public List<StudentsModels> StudentsModelses { get; set; }
+    }
+
+    public class Student
```
이건 뭐지? ㅎㅎ

raymondlee-nsuslab committed:
*피드백 수정입니다.
 - 변수명 수정
 - 약어 수정
 % httpClient Uri는 Sever to Server를 이용한 api호출에서 IIS localhost설정이 필요하여 사이먼님 도움을 받았습니다.
   그 외 코드는 인터넷 예제를 참고하였습니다.

greatyeon commented on ApiReferences/StudentReference.cs:
```diff
@@ -16,39 +16,39 @@ public List<string> GetTitles()
+                if (!response.IsSuccessStatusCode)
                 {
-                    var respErr = new HttpResponseMessage(HttpStatusCode.NotFound)
+                    var responseError = new HttpResponseMessage(HttpStatusCode.NotFound)
```
가장 흔하게 쓰는 이름을 붙이는 방법은 클래스명이 잘 지어졌다는 가정하에 클래스명을 이용하는거죠..
지금 이건 HttpResponseMessage 를 이용해서 http 에러 메시지를 만드는것이니 httpResponseMessage 라고 해줘도 될 것 같아요...

greatyeon commented on ApiReferences/StudentReference.cs:
```diff
@@ -16,39 +16,39 @@ public List<string> GetTitles()
         }
 
-        public GetRequestStudents GetStudents(Students students)
+        public GetStudentsRequest GetStudents(Students students)
```
이름 바꾸라고 한게 이거였으면... ㅎㅎ
Return 을 위한거잖아요..
그렇다면 Response 가 더 어울리겠죠...

greatyeon commented on StudentData/StudentListReturn.cs:
```diff
@@ -9,6 +9,6 @@ namespace StudentData
     {
         public int TotalRecord { get; set; }
-        public List<SchoolList> SchoolLists { get; set; }
+        public List<Students> SchoolLists { get; set; }
```
데이터에 맞는 이름을...

greatyeon reviewed (commented):
단수/복수, 그리고 데이터에 맞는 네이밍 다시 한번 정리해주세요

raymondlee-nsuslab committed:
* 변수명 수정입니다.

greatyeon reviewed (approved):
아직 좀 있네요...
일단 이거 때문에 시간을 너무 소모할 수는 없으니 일단 머지하고 이후로 계속 주의하도록 해요...
